packages/components/view-switcher/src/view-switcher.tsx

```diff
@@ -16,28 +17,44 @@ type TReactChild = {
 
 export type TViewSwitcherProps = {
   /**
-   * Indicates that the view switcher can be reduced to save space
+   * Indicates that the view switcher can be reduced to save space.
    */
   isCondensed?: boolean;
   /**
-   * Pass one or more `ViewSwitcher.Button` components
+   * Pass one or more `ViewSwitcher.Button` components.
    */
   children: ReactNode;
   /**
-   * Will be triggered whenever a `ViewSwitcher.Button` is selected. Called with the ViewSwitcherButton value
+   * Will be triggered whenever a `ViewSwitcher.Button` is selected. Called with the ViewSwitcherButton value.
+   * This function is only required when the component is controlled.
    */
   onChange?: (value: string) => void;
   /**
-   * Indicates the default selected button
+   * Passing this prop makes the component an uncontrolled component.
+   * Indicates the default selected button it is only used to as an initial state once, when the component mounts.
    */
-  defaultSelected: string;
+  defaultSelected?: string;
+  /**
+   * Passing this prop makes the component an controlled component.
+   * Controlled components also require to pass a `onChange` callback function.
+   */
+  selectedValue?: string;
```

Comment:
Should we also warn if both props are passed? Cause the `selectedValue` (controlled mode) will take precedence, right?